Storage/container/one2one.h

```diff
@@ -0,0 +1,253 @@
+    if (it == transposed_.end()) return nullptr;
+    return ImmutableOptional<T>(FromBarePointer(), it->second);
+  }
+
+  bool CanAdd(const T_RCPAIR& row_col) const {
+    return map_.find(row_col) == map_.end() && forward_.find(row_col.first) == forward_.end() &&
+           transposed_.find(row_col.second) == transposed_.end();
+  }
+  bool CanAdd(sfinae::CF<T_ROW> row, sfinae::CF<T_COL> col) const { return CanAdd(std::make_pair(row, col)); }
+
+  void operator()(const T_UPDATE_EVENT& e) {
+    const auto row = sfinae::GetRow(e.data);
+    const auto col = sfinae::GetCol(e.data);
+    DoAdd(std::make_pair(row, col), e.data);
+  }
+  void operator()(const T_DELETE_EVENT& e) { DoErase(std::make_pair(e.key.first, e.key.second)); }
+
+  template <typename T_MAP>
+  struct MapAccessor final {
+    using T_KEY = typename T_MAP::key_type;
+    const T_MAP& map_;
+
+    struct Iterator final {
+      using T_ITERATOR = typename T_MAP::const_iterator;
+      T_ITERATOR iterator_;
+      explicit Iterator(T_ITERATOR iterator) : iterator_(iterator) {}
+      void operator++() { ++iterator_; }
+      bool operator==(const Iterator& rhs) const { return iterator_ == rhs.iterator_; }
+      bool operator!=(const Iterator& rhs) const { return !operator==(rhs); }
+      sfinae::CF<T_KEY> key() const { return iterator_->first; }
+      const T& operator*() const { return *iterator_->second; }
+      const T* operator->() const { return iterator_->second; }
+    };
+
+    explicit MapAccessor(const T_MAP& map) : map_(map) {}
+
+    bool Empty() const { return map_.empty(); }
+
+    size_t Size() const { return map_.size(); }
+
+    bool Has(const T_KEY& x) const { return map_.find(x) != map_.end(); }
+
+    Iterator begin() const { return Iterator(map_.cbegin()); }
+    Iterator end() const { return Iterator(map_.cend()); }
+  };
+
+  const MapAccessor<T_FORWARD_MAP> Rows() const { return MapAccessor<T_FORWARD_MAP>(forward_); }
+
+  const MapAccessor<T_TRANSPOSED_MAP> Cols() const { return MapAccessor<T_TRANSPOSED_MAP>(transposed_); }
+
+  struct Iterator final {
+    using T_ITERATOR = typename T_WHOLE_MAP::const_iterator;
+    T_ITERATOR iterator_;
+    explicit Iterator(T_ITERATOR iterator) : iterator_(iterator) {}
+    void operator++() { ++iterator_; }
+    bool operator==(const Iterator& rhs) const { return iterator_ == rhs.iterator_; }
+    bool operator!=(const Iterator& rhs) const { return !operator==(rhs); }
+    const T_RCPAIR key() const { return iterator_->first; }
+    const T& operator*() const { return *iterator_->second; }
+    const T* operator->() const { return iterator_->second; }
+  };
+  Iterator Begin() const { return Iterator(map_.begin()); }
+  Iterator End() const { return Iterator(map_.end()); }
+
+ private:
+  void DoErase(const T_RCPAIR& row_col) {
+    forward_.erase(row_col.first);
+    transposed_.erase(row_col.second);
+    map_.erase(row_col);
+  }
+
+  void DoAdd(const T_RCPAIR& row_col, const T& object) {
```

Comment:
Perhaps:
1) another signature with `std::unique_ptr<T>&&` object to `std::move()` / `.swap()` into `map_`, and
2) another signature with `T&& object` to `std::move()` into the `unique_ptr<>` contained in the `map_` ?